tests/core/test_config_substitutor.py

```diff
@@ -242,11 +209,12 @@ class Response:
             pytest.raises(ValueError),
             None,
         ),
+        # If the regex is not matched, we return the value as-is
         (
             "secret|https://my_vault_name.vault.azure.net/secrets/my-secret/A0000000000000000000000000000000|key",
             None,
-            pytest.raises(ValueError),
-            None,
+            does_not_raise(),
+            "secret|https://my_vault_name.vault.azure.net/secrets/my-secret/A0000000000000000000000000000000|key",
```

Comment:
This was a faulty test case - normally, you'd run `substitute_config_variable`, which would call `substitute_value_from_secret_store`, which would run this string against a regex and conditionally call this Azure-specific method.

As this string doesn't match the regex, there's no way we'd even hit this method.

TLDR: The test case that was here was testing an impossible situation - we would not raise an exception so I'm changing the test case to match what would actually happen in prod.